test/cleanup.go

```diff
@@ -22,16 +22,36 @@ package test
 import (
 	"os"
 	"os/signal"
+	"sync"
 )
 
+// cleanFuncs stores all the functions to cleanup
+type cleanFuncs struct {
+	once sync.Once
+	s    *sync.Map
+}
+
+func init() {
+	cf = &cleanFuncs{s: &sync.Map{}, once: sync.Once{}}
+}
+
+var cf *cleanFuncs
+
 // CleanupOnInterrupt will execute the function cleanup if an interrupt signal is caught
 func CleanupOnInterrupt(cleanup func()) {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt)
```

Comment:
We are still registering N signals, which previously triggered all the background Go routines and allowed us to support multiple cleanups already (which all raced to `os.Exit`).

Perhaps we should register a single signal handler during `init()` and spawn a single goroutine to evaluate it, and simple allow calls to push new cleanups onto the stack?